lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -182,14 +187,42 @@ void CDataFrameTrainBoostedTreeClassifierRunner::writeOneRow(
                                 writer.Key(classValues[j]);
                                 writer.Double(shap[i](j));
                             }
-                            writer.Key(CDataFrameTrainBoostedTreeRunner::IMPORTANCE_FIELD_NAME);
+                            writer.Key(IMPORTANCE_FIELD_NAME);
                             writer.Double(shap[i].lpNorm<1>());
                         }
                         writer.EndObject();
                     }
                 }
                 writer.EndArray();
+
+                for (std::size_t i = 0; i < shap.size(); ++i) {
+                    if (shap[i].lpNorm<1>() != 0) {
+                        totalShapValues
+                            .emplace(std::make_pair(i, TVector::Zero(shap[i].size())))
+                            .first->second += shap[i].cwiseAbs();
+                    }
+                }
             });
+        writer.Key(TOTAL_FEATURE_IMPORTANCE_FIELD_NAME);
+        writer.StartArray();
+        for (const auto& item : totalShapValues) {
+            writer.StartObject();
+            writer.Key(FEATURE_NAME_FIELD_NAME);
+            writer.String(featureImportance->columnNames()[item.first]);
+            if (item.second.size() == 1) {
+                writer.Key(IMPORTANCE_FIELD_NAME);
+                writer.Double(item.second(0));
+            } else {
+                for (int j = 0; j < item.second.size() && j < numberClasses; ++j) {
+                    writer.Key(classValues[j]);
```

Comment:
This will not work for storage in ES. This index stores the information for all trained models and indexing the class names for the feature importances will not scale. 

I propose this format:
```
{
	"feature_name": "c4",
	"importance": 0.4810469375580312,
	"class_importance": [
		{
			"class_name": "foo",
			"importance": 0.24052346877901588
		},
		{
			"class_name": "baz",
			"importance": 0.19615020783390645
		},
		{
			"class_name": "bar",
			"importance": 0.04437326094510882
		}
	]
}
```

`class_importance` will be a `nested` data type that allows aggregations and searches for specific models and classnames.